cypress-io/cypress
# feat: wiring up test filters to the Debug page back end

- Closes https://github.com/cypress-io/cypress/issues/25487

### User facing changelog


n/a - merging to feature branch

### Additional details


When completing the test filtering for the new Debug page in the PR https://github.com/cypress-io/cypress/pull/25265, the backend was stubbed out with test data while the remainder of the work for the cloud integration was being done here: https://github.com/cypress-io/cypress/pull/25316/

This PR ties the work from these previous two PRs to enable the test filtering to work for each relevant run used in the new Debug page.

### Steps to test


* Record a run for a Cloud connected project into Staging or Production. 
* Open the Cypress App for that project and go to the Debug page.  
* Note the failed tests for one spec
* Click on the "Run Failures" button for that spec 
* You should be taken to the runner and only the failed tests for that Spec should run
* Select the "X" in the badge in the header where it says "X of Y tests"
* This should remove that badge and run all the tests for that spec
* Repeat with other failed specs from the Debug page to verify only the failed tests are run.

### How has the user experience changed?


UI has not changed from the original PR. 

### PR Tasks


- [ n/a] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
